community/kernel/src/main/java/org/neo4j/kernel/api/schema/index/StoreIndexDescriptor.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.schema.index;
+
+import org.neo4j.internal.kernel.api.IndexCapability;
+import org.neo4j.kernel.impl.api.index.IndexProviderMap;
+import org.neo4j.kernel.impl.api.state.ConstraintIndexCreator;
+import org.neo4j.storageengine.api.schema.SchemaRule;
+
+import static org.neo4j.internal.kernel.api.schema.SchemaUtil.idTokenNameLookup;
+
+/**
+ * Descripbes an index which is committed to the database.
+ *
+ * Adds an index id, a name, and optionally a owning constraint id to the general IndexDescriptor.
+ *
+ * Can be upgraded to a {@link CapableIndexDescriptor} by adding {@link IndexCapability} to this index,
+ * using {@link StoreIndexDescriptor#withoutCapabilities()} or {@link StoreIndexDescriptor#withCapabilities(IndexProviderMap)}
+ */
+public class StoreIndexDescriptor extends IndexDescriptor implements SchemaRule
+{
+    private final long id;
+    private final Long owningConstraintId;
+    private final String name;
+
+    // ** Copy-constructor used by sub-classes.
+    protected StoreIndexDescriptor( StoreIndexDescriptor indexDescriptor )
+    {
+        super( indexDescriptor );
+        this.id = indexDescriptor.id;
+        this.owningConstraintId = indexDescriptor.owningConstraintId;
+        this.name = indexDescriptor.name;
```

Comment:
In the copy constructor we do not need to delegate because we know that the fields of the input `StoreIndexDescriptor` will already have passed validation.